okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt

```diff
@@ -164,12 +196,27 @@ class Dispatcher() {
     val executableCalls = mutableListOf<AsyncCall>()
     val isRunning: Boolean
     synchronized(this) {
-      val i = readyAsyncCalls.iterator()
-      while (i.hasNext()) {
-        val asyncCall = i.next()
+      while (readyAsyncCalls.isNotEmpty()) {
+        val filteredCalls = readyAsyncCalls.mapNotNull {
+          if (it.callsPerHost.get() < maxRequestsPerHost) it.call else null
```

Comment:
This feels awfully close to request priorities.  Should this limit be a part of the AsycCallDispatcher functionality, allowing to execute high priority requests above the standard limit?

Reply:
I am not sure what limit you are talking about.  
If you mean `mapNotNull` part, it's for enforcing `maxRequestsPerHost`.

Reply:
Limit being maxRequestsPerHost. Essentially this limit causes priorities to break. Probably unavoidable, but I figured worth discussing whether high priority requests should override the limit?

Reply:
I don't think it shouldn't and you can see it in the (test case)[https://github.com/square/okhttp/pull/7218/files#diff-451df978a3ed8616ddb81d91165eab2cfd3bddda7bdbf7c68dcda8aac187206bR371]. I think it should work like there is a queue for every host.

Reply:
OK, makes sense.